exit.go

```diff
@@ -0,0 +1,33 @@
+package fyneselfupdate
+
+import (
+	"log"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/dialog"
+)
+
+// NewExitCallback properly exit a Fyne App. If a window is specified, it will Hide
```

Comment:
Why will it hide the window first?
Adding show seems more complexity too, why do this?
I understand that someone may want `win` to be optional, but I don't think a callback like this should be handling window lifecycle.

Reply:
There is potential strange behavior without. The application might start faster than this one close, leaving two windows visible for a short period of time which I think need to be handle by the default behavior.

I think the show was there due to some earlier test where I was hiding the windows straight after confirmation and I needed to handle the error case. I will check if that still work like expected.

Reply:
Is it worth just quitting for now, seeing how it works in the wild, and potentially refining later if required?

Reply:
I checked again, and indeed the show is a left over from a previous experience where I moved the win.Hide() inside the restart confirm callback. I had the order wrong in that function when I did the experiment. I think it is better to move back to Hide inside the restart confirm callback actually.

Reply:
Cool. Let's shift the hide/show out for now